core/controller/src/whisk/core/controller/RestAPIs.scala

```diff
@@ -115,26 +116,32 @@ protected[controller] class RestAPIVersion_v1(
      * @Idioglossia This relies on the spray routing DSL.
      * @see http://spray.io/documentation/1.2.2/spray-routing/
      */
+    private val sendCorsHeaders = respondWithHeaders(`Access-Control-Allow-Origin`(AllOrigins), `Access-Control-Allow-Headers`("Authorization", "Content-Type"))
+
     override def routes(implicit transid: TransactionId): Route = {
         pathPrefix(apipath / apiversion) {
-            pathEndOrSingleSlash {
-                complete(OK, info)
-            } ~ authenticate(basicauth) {
-                user =>
-                    namespaces.routes(user) ~
-                        pathPrefix(Collection.NAMESPACES) {
-                            actions.routes(user) ~
-                                triggers.routes(user) ~
-                                rules.routes(user) ~
-                                activations.routes(user) ~
-                                packages.routes(user)
-                        }
-            } ~ pathPrefix(swaggeruipath) {
-                getFromDirectory("/swagger-ui/")
-            } ~ path(swaggeruipath) {
-                redirect(s"$swaggeruipath/index.html", PermanentRedirect)
-            } ~ path(swaggerdocpath) {
-                getFromResource("whiskswagger.json")
+            sendCorsHeaders {
+                (pathEndOrSingleSlash & get) {
+                    complete(OK, info)
+                } ~ authenticate(basicauth) {
+                    user =>
+                        namespaces.routes(user) ~
+                            pathPrefix(Collection.NAMESPACES) {
+                                actions.routes(user) ~
+                                    triggers.routes(user) ~
+                                    rules.routes(user) ~
+                                    activations.routes(user) ~
+                                    packages.routes(user)
+                            }
+                } ~ pathPrefix(swaggeruipath) {
+                    getFromDirectory("/swagger-ui/")
+                } ~ path(swaggeruipath) {
+                    redirect(s"$swaggeruipath/index.html", PermanentRedirect)
+                } ~ path(swaggerdocpath) {
+                    getFromResource("whiskswagger.json")
+                } ~ options {
+                    complete(OK)
```

Comment:
more accurately options should respond with the allowed methods for a route - rather than a generic response; this is the goal of options (self describing routes). we should consider doing this more systematically on all the routes.